### ISSUE TITLE
handleOpenURL is not called if app was closed

### ISSUE THREAD

**Issue by dmitriyles** [EddyVerbruggen/Custom-URL-scheme#56]
I'm trying to launch my app with link on Facebook feed. If my app was opened at that time and just staying in background (`deviceready` already happened), everything is OK, `handleOpenURL` works. But if my app was closed, I get nothing. Java is trying to call `handleOpenURL`, which does not exist in that moment because my first page is not loaded. It could happen because I use splash-screen. Could you call `handleOpenURL` only after `deviceready` event?

**Comment by ohh2ahh** [comment]
@EddyVerbruggen Here's a Cordova project incl. the Xcode project (url scheme is myapp://) with the issue. Thanks for your effort!

https://dl.dropboxusercontent.com/u/10047417/Web/custom-url-scheme-issue-56.zip

**Comment by confile** [comment]
I have the same problem when the app runs in background handleOpenURL is called, but when the app was closed handleOpenURL is never called. I tested this on iPhone 5 iOS 7.1.1 CustonURL version 3.2.0

**Comment by confile** [comment]
@EddyVerbruggen is there any workaround?

**Comment by EddyVerbruggen** [comment]
Hi, for iOS, this seems to be a bug in the 3.7.0 release. Please see this issue: https://issues.apache.org/jira/browse/CB-7606

The workaround is trying the gist I linked to in that issue. Closing this because it should be solved by the Apache guys really.

**Comment by confile** [comment]
@EddyVerbruggen So this issue is fixed in cordova 4.0 right?

**Comment by EddyVerbruggen** [comment]
@confile no, it's fixed when this one is properly fixed: https://issues.apache.org/jira/browse/CB-7606 (which it isn't at the moment)

**Comment by confile** [comment]
@EddyVerbruggen the status says "Fix Version/s: 3.7.0"

**Comment by confile** [comment]
I tested both versions from here https://gist.github.com/EddyVerbruggen/3edb9f33b29d26c79ec7 The first one is not working and the second one does not work with warm start (i.e., the app is running in background).

I use an iPhone 5 iOS 7.1.1.

**Comment by EddyVerbruggen** [comment]
Yes I know it says 'Fix version 3.7.0', but clearly it isn't fixed correctly.

**Comment by mirko77** [comment]
Same thing here. It used to work on Cordova 2.9 like a breeze, now I am refactoring with Cordova 3.7 on iOS (as Apple will not accept apps without 64bits architecture support any longer) and this is not working anymore. If the app is closed the function handleOpenURL() is never called. So frustrating.

**Comment by EddyVerbruggen** [comment]
@mirko77 use iOS 3.6.3

**Comment by mirko77** [comment]
ok I will downgrade then, cheers

**Comment by confile** [comment]
Well I don't think that downgrade is a good practice. Isn't there a way to fix it?

**Comment by mirko77** [comment]
@EddyVerbruggen @confile not at the moment I am afraid

**Comment by JanMisker** [comment]
I encountered the same issue (in 3.7) and made a quick&dirty fix by editting CDViewController.m in the CordovaLib subproject.
In method `processOpenURL` change the `jsString` assignment to

    NSString* jsString = [NSString stringWithFormat:@"if (typeof handleOpenURL === 'function') { handleOpenURL(\"%@\");} else { window._handleMe = \"%@\"; }", url, url];

Then in your index.html when declaring `handleOpenURL` check whether `_handleMe` exists and call `handleOpenURL(_handleMe);`

**Comment by EddyVerbruggen** [comment]
Hi @JanMisker can you share your findings in the [Apache issue](https://issues.apache.org/jira/browse/CB-7606) please? It may be of value to guys like Shazron.

**Comment by EddyVerbruggen** [comment]
[Shazron has fixed the bug for 3.8.0](https://issues.apache.org/jira/browse/CB-7606) - I just tested it and it works fine :)

**Comment by mirko77** [comment]
Good to know, I will test it soon

**Comment by EddyVerbruggen** [comment]
Cordova-ios 3.8.0 has just been released, including [the fix for CB-7606](https://issues.apache.org/jira/browse/CB-7606). This should fix your trouble, thanks for the patience.

**Comment by Mojo90** [comment]
@EddyVerbruggen  great work for your plugins :)!
I started using this plugin now and experiencing this issue on ios.
My versions:
```
cordova: 5.3.3
Installed platforms: browser 4.0.0, ios 3.9.1
```
Everything works fine when App is in Background but my Code in handleOpenURL(url) seems not to be called when App is closed.
```
function handleOpenURL(url) {
	console.log("CALLED HANDLE OPEN URL");
	var id = getQueryVariable("release", url);
	if(id != "notfound"){
	    if(before == "releases"){
    	   	openReleaseBack(id);
    	}
    	else{
    		before = 'releases';
       		window.location.href = '#releases';
       		window.localStorage.setItem("tempReleaseOpener",id);
    	} 
    }
}
```

I say it seems to be that the function it is not called because how can I debug if it is called when App is closed? Already googled this: "attach ios simulator immediately to web inspector" but it seems that there is no solution. And I don't have an idea how to debug this because till I added the Web Inspector manually to my App everything (load of App, possible call of handleOpenURL)  is gone of course. Maybe anybody have an idea..
